src/shared_state/README.md

```diff
@@ -1,17 +1,17 @@
 # Shared state
 
-**TODO** **WIP**
+In a typical multi-user web application, clients download the app's source from a central HTTP server and rely on it to relay messages to others.
+In this [client-server model](https://en.wikipedia.org/wiki/Client%E2%80%93server_model), the server can leverage its central position of authority to ensure that all connected clients receive messages in the same order.
 
-* unlike conventional web appllications which can rely on a server to act as an authoritative source of shared state, potential conflicts between peers must be resolved on each device that runs a webxdc app
+Even though _webxdc applications_ can be used in situations where all members of a given chat are hosted by a single server, the specification does not assume this will be the case and in fact is intended to support both multi-server and fully [peer-to-peer](https://en.wikipedia.org/wiki/Peer-to-peer) modes of operation.
```

Comment:
```suggestion
Even though _webxdc applications_ can be used in situations where all members of a given chat are hosted by a single server, the specification does not assume this will be the case and in fact is intended to support both federated (multi-server) and fully [peer-to-peer](https://en.wikipedia.org/wiki/Peer-to-peer) modes of operation.   In addition, webxdc supporting messengers that implement offline-first messaging induce arbitrary delays in message delivery.  
```